.clang-tidy

```diff
@@ -0,0 +1,99 @@
+---
+Checks: >
+    bugprone-*,
```

Comment:
Many of these checks are useful and to be enabled in the future, but I disabled everything what was non-trivial to resolve at glance. 
Based on .clang-tidy from OSRM https://github.com/Project-OSRM/osrm-backend/blob/master/.clang-tidy

Reply:
We overlapped, see my comment regarding what MapBox used to have enabled and the files that we have in the repo (can probably also wait for a follow-up PR though).